### PR TITLE
feat!: extend the CLI with config- and commit- support

### PR DESCRIPTION
**Pull request by KevinDeJong-TomTom**
This commit adds the ability to:
- Manage the ruleset using a configuration file
- Check the configuration from the CLI (`cm overview`)
- Ability to use a step-by-step process for creation a commit message (`cm commit`)

This is a breaking change as significant refactoring was required

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,87 +1,159 @@
...
-    - Whenever your workflow permits it, prefer amending the original commit (or using `--fixup`) instead.
-    - If your workflow doesn't permit that, or it seems suboptimal in a given context, describe what, why and how you are changing (as mentioned before).
 
 ## Installation
 
 You can install this package with pip as follows:
 
 ```sh
-pip3 install --user --upgrade commisery
+$ pip3 install --user --upgrade commisery
```
Comment (maikelvdh) [inline README.md:30]: I would rather advise using `python3 -m pip install` sequence?

File: README.md [hunk 2]
```diff
@@ -1,87 +1,159 @@
...
 
-You can verify commit messages with the included CLI tool:
+Basic usage instructions:
+
+```
+Usage: cm [OPTIONS] COMMAND [ARGS]...
+
+  Manages conventional commit messages
+
+Options:
```
Comment (maikelvdh) [inline README.md:42]: Maybe add also version option?

File: README.md [hunk 3]
```diff
@@ -1,87 +1,159 @@
...
+
+  -v, --verbosity LVL             Either CRITICAL, ERROR, WARNING, INFO or
+                                  DEBUG
+
+  --help                          Show this message and exit.
+
+Commands:
+  check     Checks whether commit messages adhere to the Convention Commits...
+  commit    Creates a conventional commit
+  overview  Lists the accepted Conventional Commit tags and Rules (incl.
```
Comment (maikelvdh) [inline README.md:69]: Missing part of the sentence behind "incl. "?

File: README.md [hunk 4]
```diff
@@ -1,87 +1,159 @@
...
+
+> **NOTE**: *in order to remain backwards compatible, the command `commisery-verify-msg` can be used as alternative for `cm check`*
+
+### (Pre-) Commit hook
 
-After that you can use it as a hook in Git to check messages you wrote by creating a `.git/hooks/commit-msg` file with these contents:
+You can use `commisery` as a hook in Git to check messages you wrote by creating a `.git/hooks/commit-msg` file with these contents:
 ```sh
 #!/bin/sh
 exec commisery-verify-msg "$@"
```
Comment (maikelvdh) [inline README.md:111]: Why not use here `cm` also consistenyly?

File: README.md [hunk 5]
```diff
@@ -1,87 +1,159 @@
...
+                                  the default "feat" and "fix". If omitted,
+                                  uses the following list: fix, feat, build,
+                                  chore, ci, docs, perf, refactor, revert,
+                                  style, test, improvement
+
+  -l, --max-subject-length INTEGER
+                                  Maximum characters allowed in the subject of
+                                  the commit message
+
+  -d, --disable TEXT              List of commit message rules to disable. Can
```
Comment (maikelvdh) [inline README.md:55]: Maybe add also reference to the location where details of the rules can be found?

**Review by maikelvdh — DISMISSED** [review]

File: README.md (hunk 2 above)
Comment (KevinDeJong-TomTom) [reply README.md:42]: Yes, but not for this PR

File: README.md (hunk 5 above)
Comment (KevinDeJong-TomTom) [reply README.md:55]: The details can be request with `cm overview`, including a description of each rule.

File: README.md (hunk 3 above)
Comment (KevinDeJong-TomTom) [reply README.md:69]: Default behavior of `Click`, full description is visible with `cm overview --help`

File: README.md (hunk 4 above)
Comment (KevinDeJong-TomTom) [reply README.md:111]: Propose to keep the lightweight entry point for commit hook purposes

**Review by maikelvdh — DISMISSED** [review]

**Review by muggenhor — DISMISSED** [review]

**Review by maikelvdh — APPROVED** [review]

**Pull request merged**